subsys/net/ip/net_shell.c

```diff
@@ -2738,44 +2760,79 @@ static enum net_verdict _handle_ipv4_echo_reply(struct net_pkt *pkt,
 						struct net_ipv4_hdr *ip_hdr,
 						struct net_icmp_hdr *icmp_hdr)
 {
-	PR_SHELL(shell_for_ping, "Received echo reply from %s to %s\n",
+	struct net_icmpv4_echo_req *echo_repl = (struct net_icmpv4_echo_req *)(icmp_hdr + 1);
+	u32_t cycles = k_cycle_get_32() - ntohl(echo_repl->time_stamp);
+
+	PR_SHELL(shell_for_ping, "%d bytes from %s to %s: icmp_seq=%d ttl=%d time=%.2fms\n",
+		 ntohs(ip_hdr->len) - net_pkt_ipv6_ext_len(pkt) - NET_ICMPH_LEN,
 		 net_sprint_ipv4_addr(&ip_hdr->src),
-		 net_sprint_ipv4_addr(&ip_hdr->dst));
+		 net_sprint_ipv4_addr(&ip_hdr->dst),
+		 ntohs(echo_repl->sequence),
+		 ip_hdr->ttl,
+		 SYS_CLOCK_HW_CYCLES_TO_NS(cycles) / 1000000.f);
 	k_sem_give(&ping_timeout);
 	_remove_ipv4_ping_handler();
 
 	net_pkt_unref(pkt);
 	return NET_OK;
 }
 
-static int _ping_ipv4(const struct shell *shell, char *host)
+static int _ping_ipv4(const struct shell *shell,
+	char *host,
+	unsigned int count,
+	unsigned int interval)
 {
 	struct in_addr ipv4_target;
-	int ret;
+	int ret = 0;
 
 	if (net_addr_pton(AF_INET, host, &ipv4_target) < 0) {
 		return -EINVAL;
 	}
 
+	struct net_if *iface = net_if_ipv4_select_src_iface(&ipv4_target);
+
 	net_icmpv4_register_handler(&ping4_handler);
 
-	ret = net_icmpv4_send_echo_request(
-		net_if_ipv4_select_src_iface(&ipv4_target),
-		&ipv4_target,
-		sys_rand32_get(),
-		sys_rand32_get());
-	if (ret) {
-		_remove_ipv4_ping_handler();
-	} else {
-		PR("Sent a ping to %s\n", host);
+	PR("PING %s\n", host);
+
+	for (int i = 0; i < count; ++i) {
+
+		ret = net_icmpv4_send_echo_request(
+				iface,
```

Comment:
move iface right after the preceding '('

and indent properly the parameters below '(' + 1 space

Reply:
moved it around